WSLGd/ProcessMonitor.cpp

```diff
@@ -26,8 +26,21 @@ int wslgd::ProcessMonitor::LaunchProcess(
     if (childPid == 0) try {
         // Construct a null-terminated argument array.
         std::vector<const char*> arguments;
-        for (auto &arg : argv) {
-            arguments.push_back(arg.c_str());
+        if (argv[0].compare("/init") != 0) {
+            for (auto &arg : argv) {
+                arguments.push_back(arg.c_str());
+            }
+        } else {
+            for (int i = 0; i < argv.size(); i++) {
+                arguments.push_back(argv[i].c_str());
+                if (i == 1) {

```

Comment:
I'm not sure if it's better to do this here, or let caller to insert it., any suggestion?

Reply:
I have a small preference for this being done by the caller since it follows the exec* convention more closely

Reply:
Changed to insert the exe argument at caller